collector/CMakeLists.txt

```diff
@@ -68,8 +68,8 @@ target_link_libraries(collector_lib z ssl crypto CURL::libcurl bpf)
 add_executable(collector collector.cpp)
 target_link_libraries(collector collector_lib)
 
-if(NOT DEFINED ENV{DISABLE_PROFILING})
-	target_link_libraries(collector profiler tcmalloc)
+if(NOT DEFINED ENV{DISABLE_PROFILING} AND (NOT DEFINED ENV{USE_VALGRIND} OR NOT "$ENV{USE_VALGRIND}"))
```

Comment:
Looks like we can set the env variable `RUNNING_ON_VALGRIND=1` to prevent gperftools from changing the malloc implementation, allowing valgrind to work correctly, we might want to remove everything after the `AND` in this condition.

Reply:
Ended up with a different approach, basically disabling profiling if we are compiling collector with asan or with the `USE_VALGRIND` flag (though this last one might be unnecessary, since we can just disable profiling manually).